homeassistant/components/shelly/binary_sensor.py

```diff
@@ -200,9 +210,13 @@ def is_on(self) -> bool:
 class RestBinarySensor(ShellyRestAttributeEntity, BinarySensorEntity):
     """Represent a REST binary sensor entity."""
 
+    entity_description: RestBinarySensorDescription
+
     @property
-    def is_on(self) -> bool:
+    def is_on(self) -> bool | None:
         """Return true if REST sensor state is on."""
+        if self.attribute_value is None:
+            return None
```

Comment:
This is a slight change in logic, as binary sensors can return `None` (see #60193)